### PR TITLE
coding of games added

### PR DESCRIPTION
**Pull request by kwons90**
Codification of games is added. How it's working:
- it checks whether the session has a gameSessionID. If so, I do nothing
- If there isn't one, I create a gameSession and attach it to the session instance
- I generate a code, and check if Session Model has an instance with this code. If so, I generate another
- I attach this code to the GameSession instance and attach it to the Session instance

### CODE REVIEW

File: src/client/store/reducers/gameReducer.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    active: false,
+  }
+  
+  const gameReducer=(state=initialGameState, action)=>{
+    switch(action.type){
+      case types.NEW_GAME:
+        return {
+          ...state,
+          currentId: action.payload.id,
+          active: true,
```
Comment (jspicuzza21) [inline src/client/store/reducers/gameReducer.js:16]: I think that this would have to be action.payload.game

File: src/server/api/index.js [hunk 2]
```diff
@@ -13,14 +14,26 @@ const {
...
 const DIST_PATH = join(__dirname, '../../../dist');
 
 app.use(cookieParser());
 
+
+const codeGenerator = () => {
+  let result = '';
+  const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789';
+  const charactersLength = characters.length;
+  for (let i = 0; i < 5; i += 1) {
```
Comment (jspicuzza21) [inline src/server/api/index.js:30]: This should go in a utils file for readability

File: src/server/api/index.js [hunk 3]
```diff
@@ -48,6 +61,27 @@ app.use(async (req, res, next) => {
...
 
+// assign games if they don't have
+app.use(async (req, res, next) => {
+  const session = await Session.findOne({ where: {id:req.session_id}})
+  // console.log(session.gameSessionId)
+  if(!session.gameSessionId) {
+    let newCode = codeGenerator()
+    console.log(newCode)
+    let check = await GameSession.findOne({ where: { code: newCode } });
+    while (check) {
```
Comment (jspicuzza21) [inline src/server/api/index.js:72]: Probably could have just used the gameSession ID here but it's the same in the end.

File: src/server/api/index.js (hunk 2 above)
Comment (kwons90) [reply src/server/api/index.js:30]: This caused issues when I tried to do that -can clean it up later

File: src/server/api/index.js (hunk 3 above)
Comment (kwons90) [reply src/server/api/index.js:72]: Ok

**Comment by jspicuzza21** [comment]
Looks good

**Review by jspicuzza21 — APPROVED** [review]

**Pull request merged**